platforms/posix/src/px4_layer/px4_posix_tasks.cpp

```diff
@@ -140,6 +143,10 @@ px4_task_t px4_task_spawn_cmd(const char *name, int scheduler, int priority, int
 		len += strlen(p) + 1;
 	}
 
+	// We add one argument for the name as argv[0].
+	++argc;
```

Comment:
Placing this right after `len += strlen(name) + 1;` would IMO make more sense.

Otherwise looks good.